components/promise-component.js

```diff
@@ -0,0 +1,66 @@
+// promise-component.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+module.exports = function(def){
+    if (_.isEmpty(def.inPorts)) throw Error("Missing inPorts");
+    if (def.resolvePort && !_.isString(def.resolvePort.name))
+        throw Error("Missing resolvePort name");
+    if (def.rejectPort && !_.isString(def.rejectPort.name))
+        throw Error("Missing rejectPort name");
+    var resolvePort = def.resolvePort || _.defaults({
+        name: 'out'
+    }, _.propertyOf(def.outPorts)('out')) || {
+        name: 'out',
+        description: "Resolved result of the updater",
+        datatype: 'object',
+        addressable: false,
+        buffered: false,
+        required: false,
+        caching: false
+    };
+    var rejectPort = def.rejectPort || _.defaults({
+        name: 'error'
+    }, _.propertyOf(def.outPorts)('error')) || {
+        name: 'error',
+        description: "Rejected error of the updater",
+        datatype: 'object',
+        addressable: false,
+        buffered: false,
+        required: false,
+        caching: false
+    };
+    var inPorts = _.mapObject(def.inPorts, function(port, name){
```

Comment:
Right.  Was on the wrong line with my comment.  The following comment is what I meant.  No issue here.